addons/road-generator/nodes/road_branch.gd

```diff
@@ -0,0 +1,45 @@
+@tool
+@icon("res://addons/road-generator/resources/road_branch.png") #TODO: add real icon
+## Road Branch is a node in a road network. Road branches are connected via Road paths.
+## A Road Branch may contain a Road Intersection which would dictate crossing behavior.
+
+class_name RoadBranch
+extends Node3D
+
+var _entry_paths : Array[RoadPath] = []
+@export var entry_paths : Array[RoadPath] :
+	get:
+		return _entry_paths
+		
+var _exit_paths : Array[RoadPath] = []
+@export var exit_paths : Array[RoadPath] :
+	get:
+		return _exit_paths
+
+func _ready():
+	pass
+
+
+# Workaround for cyclic typing
+func is_road_branch() -> bool:
+	return true
```

Comment:
fwiw this might be no longer needed, a relic of the 3.x era! (but, to be seen later I suppose)